### PR TITLE
[1.2.x] Fixes security vulnerabilities related mime module

### PR DESCRIPTION
**Pull request by TharmiganK**
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/4366

### CODE REVIEW

File: stdlib/mime/src/main/java/org/ballerinalang/mime/util/MultipartDataSource.java [hunk 1]
```diff
@@ -111,6 +111,12 @@ private void serializeBodyPart(OutputStream outputStream, String parentBoundaryS
             writeFinalBoundaryString(writer, parentBoundaryString);
         } catch (IOException e) {
             log.error("Error occurred while writing body parts to outputstream", e.getMessage());
+        } finally {
```
Comment (gabilang) [inline stdlib/mime/src/main/java/org/ballerinalang/mime/util/MultipartDataSource.java:114]: No need for the `finally` block and explicit `close()` method call here, if we add the new `writer` object creation to the try block (by using the try-with-resources construct)

Comment (TharmiganK) [reply stdlib/mime/src/main/java/org/ballerinalang/mime/util/MultipartDataSource.java]: Fixed via [83de62e](https://github.com/ballerina-platform/ballerina-lang/pull/40263/commits/83de62e8ebbab0c106a2a117fd7cd5b039e81317)

File: stdlib/mime/src/main/java/org/ballerinalang/mime/util/MultipartDataSource.java [hunk 2]
```diff
@@ -86,10 +86,9 @@ public void serialize(OutputStream outputStream) {
      */
     private void serializeBodyPart(OutputStream outputStream, String parentBoundaryString,
                                    ObjectValue parentBodyPart) {
-        final Writer writer = new BufferedWriter(new OutputStreamWriter(outputStream, Charset.defaultCharset()));
         ArrayValue childParts = parentBodyPart.getNativeData(BODY_PARTS) != null ?
                 (ArrayValue) parentBodyPart.getNativeData(BODY_PARTS) : null;
-        try {
+        try (final Writer writer = new BufferedWriter(new OutputStreamWriter(outputStream, Charset.defaultCharset()))){
```
Comment (gabilang) [inline stdlib/mime/src/main/java/org/ballerinalang/mime/util/MultipartDataSource.java:91]: You need to instantiate the resources one by one. Refer https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Comment (gabilang) [reply stdlib/mime/src/main/java/org/ballerinalang/mime/util/MultipartDataSource.java:91]: A good read on the issues with inner resource instantiation https://dzone.com/articles/carefully-specify-multiple-resources-in-single-try

Comment (TharmiganK) [reply stdlib/mime/src/main/java/org/ballerinalang/mime/util/MultipartDataSource.java]: Fixed via https://github.com/ballerina-platform/ballerina-lang/pull/40263/commits/01339fd5fc2551466cbb45cb8beb5e2a96f67db0

**Review by gabilang — APPROVED** [review]
LGTM

**Pull request merged**